### PR TITLE
Remove Direct Runner "doneness" configuration

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by tgroh** [comment]
@jkff

File: runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java [hunk 1]
```diff
@@ -422,9 +422,7 @@ public MetricResults metrics() {
      * {@link org.apache.beam.sdk.PipelineResult.State#FAILED}.
      *
      * <p>NOTE: if the {@link Pipeline} contains an {@link IsBounded#UNBOUNDED unbounded}
-     * {@link PCollection}, and the {@link PipelineRunner} was created with
-     * {@link DirectOptions#isShutdownUnboundedProducersWithMaxWatermark()} set to false,
-     * this method will never return.
+     * {@link PCollection} this method will never return.
```
Comment (jkff) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java]: This is not true, right? The whole paragraph should be deleted.

File: runners/direct-java/src/main/java/org/apache/beam/runners/direct/EvaluationContext.java [hunk 2]
```diff
@@ -406,33 +405,11 @@ void forceRefresh() {
    * <p>If the provided transform produces only {@link IsBounded#BOUNDED}
    * {@link PCollection PCollections}, returns true if the watermark is at
    * {@link BoundedWindow#TIMESTAMP_MAX_VALUE positive infinity}.
-   *
-   * <p>If the provided transform produces any {@link IsBounded#UNBOUNDED}
-   * {@link PCollection PCollections}, returns the value of
-   * {@link DirectOptions#isShutdownUnboundedProducersWithMaxWatermark()}.
    */
   public boolean isDone(AppliedPTransform<?, ?, ?> transform) {
     // if the PTransform's watermark isn't at the max value, it isn't done
```
Comment (jkff) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/EvaluationContext.java]: Remove the double negation?

File: runners/direct-java/src/main/java/org/apache/beam/runners/direct/EvaluationContext.java [hunk 3]
```diff
@@ -406,33 +405,11 @@ void forceRefresh() {
    * <p>If the provided transform produces only {@link IsBounded#BOUNDED}
```
Comment (jkff) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/EvaluationContext.java]: Should this comment be updated too? I think it's not necessary to restrict this to bounded collections.

File: runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java (hunk 1 above)
Comment (tgroh) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java]: Rm'd.

File: runners/direct-java/src/main/java/org/apache/beam/runners/direct/EvaluationContext.java (hunk 2 above)
Comment (tgroh) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/EvaluationContext.java]: Done.

File: runners/direct-java/src/main/java/org/apache/beam/runners/direct/EvaluationContext.java (hunk 3 above)
Comment (tgroh) [inline runners/direct-java/src/main/java/org/apache/beam/runners/direct/EvaluationContext.java]: Done.

**Pull request closed without merging**